src/skgmsh/__init__.py

```diff
@@ -296,6 +288,65 @@ def frontal_delaunay_2d(  # noqa: C901, PLR0912
     return mesh.remove_cells(ind)
 
 
+def generate_mesh(dim) -> pv.UnstructuredGrid:
+    """
+    Generate a mesh of the current model.
+
+    Parameters
+    ----------
+    dim : int
+        Mesh dimension.
+
+    Returns
+    -------
+    pyvista.UnstructuredGrid
+        Generated mesh.
+
+    """
+    import numpy as np
+
+    gmsh_to_pyvista_type = {
+        1: pv.CellType.LINE,
+        2: pv.CellType.TRIANGLE,
+        3: pv.CellType.QUAD,
+        4: pv.CellType.TETRA,
+        5: pv.CellType.HEXAHEDRON,
+        6: pv.CellType.WEDGE,
+        7: pv.CellType.PYRAMID,
+        15: pv.CellType.VERTEX,
+    }
+
+    try:
+        gmsh.model.mesh.generate(dim)
+        node_tags, coord, _ = gmsh.model.mesh.getNodes()
+        element_types, element_tags, element_node_tags = gmsh.model.mesh.getElements()
+
+        # Points
+        assert (np.diff(node_tags) > 0).all()
```

Comment:
Not sure if this is ever False. ``pygmsh`` is checking whether tags are sorted, so I left it.